samza-core/src/main/java/org/apache/samza/processor/SamzaContainerController.java

```diff
@@ -77,6 +80,7 @@ public SamzaContainerController(
     } else {
       this.containerShutdownMs = containerShutdownMs;
     }
+    this.lifeCycleAwares = lifeCycleAwares;
```

Comment:
> Shutdown means either user triggered shutdown or finite input streams. Failure happens when exception thrown from the container thread.
@xinyuiscool Can you please capture this in the documentation? 

A more general question: If this lifecycle callback is not user-facing, what kind of interactions does it expects from StreamProcessor? The fact that you are directly injecting the lifecycle handler by-passing the StreamProcessor interface is extremely confusing and blurs the contract between the StreamProcessor and ApplicationRunner. Can you please clarify the contract here?